content/en/docs/contribute/generate-ref-docs/kubernetes-components.md

```diff
@@ -17,66 +17,48 @@ reference documentation for tools and components in the
 
 * You need to have this software installed:
 
+    * [Python 2.7.16](https://www.python.org/downloads/)
```

Comment:
hello @aimeeu . I looked through the changes found a few things to tidy up. I made note of some suggested edits.

Line 8: Possible edit of the overview:
This page shows how to use the `update-imported-docs` tool to generate documentation for the Kubernetes tools and components. The `update-imported-docs` Python script is located in the `kubernetes/website/update-imported-docs/`directory.

Line 20: Is it possible to use Python 3? I would assume so.
Line 22: Is it necessary to list the version of Kubernetes here? Golang 1.13 ...
Line 27: Is it true that you need to unset/not set the GOPATH. I would have to confirm this. If I recall correctly, the script first clones the named repo into the tmp dir, then sets the GOPATH, then runs the series of commands in the configuration file. If it is true, I would make this instruction a separate list item.
Line 31: fix the link, [Work from a local clone](/docs/contribute/intermediate/#work-from-a-local-clone). This link could be shortened in the corresponding file.
Line 37: possibly be more specific: Getting the kubernetes/website repository
Line 39: What happens if you already have a clone of your docs fork? I think we assume that the reader has a clone as they are looking at the docs and the update script?
Line 43: What is: `$github.com/website`.
Line 49: mkdir github.com? Where on the file system?
Line 53: Possibly add this link, from the [Kubernetes source code](https://github.com/kubernetes/kubernetes) repository
Line 61: Clean up the intro sentence as mentioned (remove: The tool consists of). Add a list item that indicates that the script takes an argument, a YAML configuration file. Add another list item about that the release version in the configuration should match the version of the current release.
Line 70: Possible edit: Updates `kubectl` command links in the generated file, `kubectl`.md.
Lines 76 - 101. I agree. The information is somewhat extraneous.
Line 112: You may want to add information about updating the release number in the configuration file.
Line 175: Fixing links: What does release.yml actually do? Is it *only* used to update the release notes?
Line 208: Possibly add a new header to build the gen'ed files:
If this page is like a playbook for generating ref docs, then before creating the pull request, the user should build the documentation locally and review the changes :smile:

Reply:
Thanks @kbhawkey  for your awesome feedback - very much appreciated! 

Line 8: kinda funny - I went back and forth on using "Python script" vs "tool" and opted to keep the original wording

Line 20: I could not run the script with Python 3 and PyYAML 5.1.x (which is automatically installed using `pip install pyyaml`).  That's why I started on PR #16681 To test with Python 2.7.16, I created a virtual environment and installed Python 2.7.16 and PyYAML 3.13 inside it - I figured those versions were close to what was available when the script was written.  

Line 22: Yes, I believe it is. I ran into issues trying to run the script when I had Go 1.13 installed, and Kubernetes development does not support 1.13.  Or do you mean to link to the list of the supported Go version for each Kubernetes version? 

Line 27: recalling our Slack conversation, I think both you and I unset our GOPATH before running the script. I agree - more testing is needed since the script does set the GOPATH. My concern is that before I commented out my existing GOPATH in my .bashrc, the Go code executed by the Python script could not find the cloned repos, making me think that my GOPATH was being found before the GOPATH set by the script

Line 39: I assume the reader is experienced enough to know NOT to clone the website repo if it already exists locally. Maybe omit this bit altogether?  Perhaps specifying a certain level of knowledge as a prerequisite... these instructions assume you know how to create a website pull request

Line 49: I think this section should be omitted - I tried to update because the kubernetes/website repo does not need to be cloned into an existing GOPATH; yet the doc refers to <website_root> further down so tried to give an example; the location of kubernetes/website is not relevant - I've run the script now with kubernetes/website cloned into different directories. 

Line 175: I have no idea what the release.yml file does or why it's included - maybe @tenqm knows?

Reply:
Wondering if we have any error outputs when running the script against python3?

Reply:
Regarding the `release.yml` file, it is used as an input to the script so that the release notes can be auto generated/imported from upstream rep.

```
./update-imported-docs release.yml
```

Reply:
I had a moment to test the script with python3. I kept my GOPATH set (not including the dynamically gen'ed tmp location). With a few updates to the script, the script ran w/o error. I will create a PR with the changes to the script.
I did not set up a virtualenv.

Reply:
@kbhawkey are you able to update my PR  [WIP 16681](https://github.com/kubernetes/website/pull/16681), which I created for updating the script to run under Python 3 and PyYAML 5.1?